### PR TITLE
Fix: Email Parsing for html fix

### LINKED ISSUE #3539: chore: Improve email reply parsing
At the moment chatwoot uses a homegrown logic to parse message content in emails that need to be appended to the conversation. But this function tends to break down for specific email payloads. Since there are gems out there that solve this let's replace this logic with an existing solution instead. 

ref: https://github.com/chatwoot/chatwoot/blob/dcdeaa79543939ccf0a4be50ce4bcb6811cb0795/app/presenters/mail_presenter.rb#L105 

## Proposed solution 
use  [email_reply_trimmer](https://github.com/discourse/email_reply_trimmer) gem from discourse. 

Also used by GitLab 

https://gitlab.com/gitlab-org/gitlab/-/commit/f7b0692912e0679a3e2e77b2d1bfaf305fba473a
https://gitlab.com/gitlab-org/gitlab/-/commit/a3bb2463c41124d6d7f5927bbff578a4991c15cb

## Alternative solutions

https://github.com/github/email_reply_parser ( seems to be abandoned ) 
https://github.com/mailgun/talon (licensing conflicts ) ?

### PR DESCRIPTION
**Pull request by tejaswinichile**
## Description

Email Parsing fix.

Fixes #3539 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

The HTML data was getting converted to plain text. Suppose we have a bold text in the email or an HTML table the previous fix was removing all the HTML tags and was showing the text without HTML code.
This fix will show the HTML code with quoted text and image/attachments as it is in an email.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tejaswinichile** [comment]
email_test

Fixes the issue.

**Comment by tejaswinichile** [comment]
@sojan-official Please check this fix.

**Comment by sojan-official** [comment]
@tejaswinichile  Let's do a full QA for the following cases and ensure we have test cases for the following

## conversation continuity 

- [ ] Text replies in Conversation continuity ( Chatwoot message summary in quoted text )
- [ ] Text replies in Conversation continuity ( broken quoted text )
- [ ] HTML replies in conversation continuity - ( should render the HTML properly ) 
- [ ] HTML replies in conversation continuity - broken quoted text

## email channel 

- [ ] Incoming HTML email ( marketing emails, LinkedIn notifications etc )
- [ ] Incoming HTML email with broken HTML 
- [ ] Reply to HTML emails in email channel
- [ ] Reply to HTML emails with broken quoted text  

## Edge cases
- Multi-part email with empty content

cc: @pranavrajs

File: app/presenters/mail_presenter.rb [hunk 1]
```diff
@@ -8,30 +8,47 @@ def initialize(mail, account = nil)
...
-      full: @decoded_text_content,
-      reply: extract_reply(@decoded_text_content)[:reply],
-      quoted: extract_reply(@decoded_text_content)[:quoted_text]
+      full: select_body(text_part),
+      reply: @decoded_text_content,
+      quoted: body.force_encoding(encoding).encode('UTF-8')
     }
   end
 
+  def select_body(mail_part)
```
Comment (sojan-official) [inline app/presenters/mail_presenter.rb:29]: can we add some comments in this method on how it works ?

File: app/presenters/mail_presenter.rb [hunk 2]
```diff
@@ -8,30 +8,47 @@ def initialize(mail, account = nil)
...
+    return '' unless mail_part
+
+    decoded = encode_to_unicode(mail_part.decoded)
+
+    if mail.text_part
+      decoded
+    elsif (mail.content_type || '').include? 'text/html'
+      ::HtmlParser.parse_reply(decoded)
```
Comment (sojan-official) [inline app/presenters/mail_presenter.rb:37]: if do `parse_reply` on HTML emails will it strip off all the HTML and convert it into a text email ?

Comment (tejaswinichile) [reply app/presenters/mail_presenter.rb:37]: No, it parses the HTML tags and applies them. It doesn't convert it to the text.

**Comment by tejaswinichile** [comment]
img


Html image loading test done!

**Comment by tejaswinichile** [comment]
attachment-without-content

Attachment without mail content working fine.

**Comment by tejaswinichile** [comment]
@pranavrajs / @sojan-official I just need one more eye of QA on this PR, I have tested everything on local and one of our clients is also using these changes for the HTML email. I couldn't set the channel upon the staging instance, please test and merge whenever you guys get the time.

**Pull request merged**